libbeat/idxmgmt/std_test.go

```diff
@@ -197,7 +215,59 @@ func TestDefaultSupport_BuildSelector(t *testing.T) {
 	}
 }
 
-func TestDefaultSupport_TemplateHandling(t *testing.T) {
+func TestIndexManager_VerifySetup(t *testing.T) {
+	for name, setup := range map[string]struct {
+		tmpl, ilm         bool
+		loadTmpl, loadILM LoadMode
+		ok                bool
+		warn              string
+	}{
+		"load template with ilm without loading ilm": {
+			ilm: true, tmpl: true, loadILM: LoadModeDisabled,
+			warn: "whithout loading ILM policy and alias",
+		},
+		"load ilm without template": {
+			ilm: true, loadILM: LoadModeUnset,
+			warn: "without loading template is not recommended",
+		},
+		"template disabled but loading enabled": {
+			loadTmpl: LoadModeEnabled,
+			warn:     "loading not enabled",
+		},
+		"ilm disabled but loading enabled": {
+			loadILM: LoadModeEnabled, tmpl: true,
+			warn: "loading not enabled",
+		},
+		"ilm enabled but loading disabled": {
+			ilm: true, loadILM: LoadModeDisabled,
+			warn: "loading not enabled",
+		},
+		"template enabled but loading disabled": {
+			tmpl: true, loadTmpl: LoadModeDisabled,
+			warn: "loading not enabled",
+		},
+		"everything enabled": {
+			tmpl: true, loadTmpl: LoadModeUnset, ilm: true, loadILM: LoadModeUnset,
+			ok: true,
+		},
+	} {
+		t.Run(name, func(t *testing.T) {
+			cfg, err := common.NewConfigFrom(common.MapStr{
+				"setup.ilm.enabled":      setup.ilm,
+				"setup.template.enabled": setup.tmpl,
+			})
+			require.NoError(t, err)
+			support, err := MakeDefaultSupport(ilm.StdSupport)(nil, beat.Info{}, cfg)
+			require.NoError(t, err)
+			manager := support.Manager(newMockClientHandler(), nil)
+			ok, warn := manager.VerifySetup(setup.loadTmpl, setup.loadILM)
+			assert.Equal(t, setup.ok, ok)
+			assert.Contains(t, warn, setup.warn)
+		})
```

Comment:
when using the mockClientHandler always run `assertInvariants` at the end of an test. E.g. for `VerifySetup` no ops should have been recorded (adds some behavioral testing).